src/StatementFor.cpp

```diff
@@ -146,6 +149,8 @@ make_random_array_control(unsigned int bound, int &init, int &limit, int &incr,
 		// decrement, start near last index
 		init = pure_rnd_flipcoin(50) ? (bound) : (bound - pure_rnd_upto(bound/2));
 		limit = pure_rnd_flipcoin(50) ? 0 : pure_rnd_upto(bound/2);
+ //       limit =-100000000;
+        limit = oob_flipcoin ? -100000000 : pure_rnd_flipcoin(50) ? 0 : pure_rnd_upto(bound/2);
```

Comment:
Shouldn't we set the limit to a big enough number to trigger OOB?

Reply:
Thanks for the review. The -100000000 limit was set for decreasing array accesses to force OOB. However, I've now decreased the limit now since I had noticed that due to nested loop generation, there is a chance that the generated program would get stuck in a loop.

Reply:
Ok, make sense.